src/services/TestResultsService.ts

```diff
@@ -545,27 +542,27 @@ export class TestResultsService {
     return missingFields;
   }
 
-  private validateMandatoryTestResultFields(payload: ITestResultPayload): string[] {
-    const missingMandatoryFields: string[] = [];
-    if (payload.testTypes.some((testType: TestType) => testType.testResult !== TEST_RESULT.ABANDONED) && payload.testStatus === TEST_STATUS.SUBMITTED) {
-      if (!payload.countryOfRegistration) {
-        missingMandatoryFields.push(ERRORS.CountryOfRegistrationMandatory);
-      }
-      if (!payload.euVehicleCategory) {
-        missingMandatoryFields.push(ERRORS.EuVehicleCategoryMandatory);
-      }
+  private addMandatoryTestResultFieldsValidation(payload: ITestResultPayload, validationSchema: Joi.ObjectSchema | null) {
+    let updatedValidationSchema: Joi.ObjectSchema | null = Object.create(validationSchema);
+    if (payload.testStatus === TEST_STATUS.SUBMITTED && updatedValidationSchema !== null && payload.testTypes &&
+      payload.testTypes.some((testType: TestType) => testType.testResult !== TEST_RESULT.ABANDONED)) {
+
+      updatedValidationSchema = updatedValidationSchema.keys({
+        euVehicleCategory: Joi.any().only(["m1", "m2", "m3", "n1", "n2", "n3", "o1", "o2", "o3", "o4"]).required(),
```

Comment:
Are we planning to to do an iteration to add the values for motorcycle e.g. `"l1e-a", "l1e", "l2e"`

Reply:
Yes, these bug fixes are not related to specialist tests. If specialist tests will be merged before this PR, I will have to add the categories related to the newly introduced vehicle types.